client/daemon/peer/piece_manager.go

```diff
@@ -385,7 +385,7 @@ singleDownload:
 			}
 		}
 		srcErr := &errordetailsv1.SourceError{
-			Temporary: response.Temporary != nil && response.Temporary(),
+			Temporary: response.Temporary == nil || response.Temporary(),
```

Comment:
Add repose.Temporary to https://github.com/dragonflyoss/Dragonfly2/blob/main/pkg/source/clients/httpprotocol/http_source_client.go#L263